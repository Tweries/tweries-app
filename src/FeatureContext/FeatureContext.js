// TODO: move it to a separate libary

import { createContext, useContext } from 'react';

const FeatureContext = createContext();

export default FeatureContext;

export function setFeatures(features) {
  return {
    active: feature => {
      return features.indexOf(feature) !== -1;
    }
  };
}

export function useFeature() {
  return useContext(FeatureContext);
}
