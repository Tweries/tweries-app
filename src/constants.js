export const BASE_URL = 'https://china-musk-api.herokuapp.com'; // 'https://china-musk-api.herokuapp.com' | 'http://localhost:9000'

export const FEATURE_V1 = 'FEATURE_V1';
export const SPACE_AFTER_PUNCTUATION_V1 = 'SPACE_AFTER_PUNCTUATION_V1';

export const LINEFEED = ';;';
export const NEWLINE = '\n';
