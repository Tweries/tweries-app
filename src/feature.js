import { setFeatures } from './FeatureContext/FeatureContext';

import {
  EDITABLE_TWEETSTORM_V1,
  EDITABLE_TWEETSTORM_COPY_V1,
  FEATURE_V1,
  READONLY_TWEETSTORM_V2
} from './constants';

const features = [
  EDITABLE_TWEETSTORM_V1,
  EDITABLE_TWEETSTORM_COPY_V1,
  FEATURE_V1,
  READONLY_TWEETSTORM_V2
];

const feature = setFeatures(features);

export default feature;
