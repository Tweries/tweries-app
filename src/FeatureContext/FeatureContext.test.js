import { render } from '@testing-library/react';
import React from 'react';
import { FEATURE_V1 } from '../constants';
import feature from '../feature';
import FeatureContext, { useFeature } from './FeatureContext';

function Paragraph() {
  const feature = useFeature();
  return feature.active(FEATURE_V1) ? <p>YATTA</p> : <p>Oh Noes!</p>;
}

test(`${FEATURE_V1} is active`, () => {
  const { container } = render(
    <FeatureContext.Provider value={feature}>
      <Paragraph />
    </FeatureContext.Provider>
  );

  expect(container).toMatchSnapshot();
});
