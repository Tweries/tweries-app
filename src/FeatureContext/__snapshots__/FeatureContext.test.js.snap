// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FEATURE_V1 is active 1`] = `
<div>
  <p>
    YATTA
  </p>
</div>
`;
